tests/versioning.go

```diff
@@ -5022,6 +5024,13 @@ func (s *VersioningIntegSuite) waitForWorkflowBuildId(
 	)
 }
 
+// validateWorkflowBuildIds gets the specified workflow execution from visibility and validates that the BuildIds
+// SearchAttribute has the expected characteristics.
+//   - expectedBuildId: the currently-assigned build id for the workflow execution (only present in newVersioning)
+//   - expectedStampBuildId: the most recent version of a worker that processed a task for that workflow execution
+//
+// If expectedBuildId != "" and newVersioning == true, waitForWorkflowBuildId should be called before this to prevent
```

Comment:
The reason I'm calling this `waitForWorkflowBuildId` in `firstWorkflowTaskAssignmentSpooled` is that I'm checking wf build ID before the task is dispatched (when worker is not running). That relies on the history task processing timing because MS build ID is updated after the transfer task is sent to Matching and it is backlogged.

MS build ID is also updated when the task is dispatched (when record*TaskStarted is called). So in tests that wait for task dispatch via other means like waiting for the `timedoutTask` channel for example, calling `waitForWorkflowBuildId` should not add any value. Still I see I called it in `firstWorkflowTaskAssignmentSyncMatch` and `testWorkflowTaskRedirectInRetry` that already are waiting for channels. Not sure if I had a reason or it was a mistake. Wish I had put a note there.